code/datums/status_effects/debuffs/terrified.dm

```diff
@@ -0,0 +1,156 @@
+		return
+
+	if(terror_buildup >= TERROR_FEAR_THRESHOLD) //The onset, minor effects of terror buildup
+		owner.adjust_dizzy_up_to(10 SECONDS * delta_time, 10 SECONDS)
+		owner.adjust_stutter_up_to(10 SECONDS * delta_time, 10 SECONDS)
+		owner.adjust_jitter_up_to(10 SECONDS * delta_time, 10 SECONDS)
+
+	if(terror_buildup >= TERROR_PANIC_THRESHOLD) //If you reach this amount of buildup in an engagement, it's time to start looking for a way out.
+		owner.playsound_local(get_turf(owner), 'sound/health/slowbeat.ogg', 40, 0, channel = CHANNEL_HEARTBEAT, use_reverb = FALSE)
+		owner.add_fov_trait(id, FOV_270_DEGREES) //Terror induced tunnel vision
+		owner.adjust_eye_blur_up_to(10 SECONDS * delta_time, 10 SECONDS)
+		if(prob(5)) //We have a little panic attack. Consider it GENTLE ENCOURAGEMENT to start running away.
+			freak_out(PANIC_ATTACK_TERROR_AMOUNT)
+			owner.visible_message(span_warning("[owner] drops to the floor for a moment, clutching their chest."), span_alert("Your heart lurches in your chest. You can't take much more of this!"), span_hear("You hear a grunt."))
+	else
+		owner.remove_fov_trait(id, FOV_270_DEGREES)
+
+	if(terror_buildup >= TERROR_HEART_ATTACK_THRESHOLD) //You should only be able to reach this by actively terrorizing someone
+		owner.visible_message(span_warning("[owner] clutches [owner.p_their()] chest for a moment, then collapses to the floor."), span_alert("The shadows begin to creep up from the corners of your vision, and then there is nothing..."), span_hear("You hear something heavy collide with the ground."))
+		var/datum/disease/heart_failure/heart_attack = new(src)
+		heart_attack.stage_prob = 2 //Advances twice as fast
+		owner.ForceContractDisease(heart_attack)
+		owner.Unconscious(20 SECONDS)
+		qdel(src) //Victim passes out from fear, calming them down and permenantly damaging their heart.
+
+/datum/status_effect/terrified/get_examine_text()
+	if(terror_buildup > DARKNESS_TERROR_CAP) //If we're approaching a heart attack
+		return span_boldwarning("[owner.p_they(TRUE)] [owner.p_are()] seizing up, about to collapse in fear!")
+
+	if(terror_buildup >= TERROR_PANIC_THRESHOLD)
+		return span_boldwarning("[owner] is visibly trembling and twitching. It looks like [owner.p_theyre()] freaking out!")
+
+	if(terror_buildup >= TERROR_FEAR_THRESHOLD)
+		return span_warning("[owner] looks very worried about something. [owner.p_are(TRUE)] [owner.p_they()] alright?")
+
+	return span_notice("[owner] looks rather anxious. [owner.p_they(TRUE)] could probably use a hug...")
+
+/// If we get a hug from a friend, we calm down! If we get a hug from a nightmare, we FREAK OUT.
+/datum/status_effect/terrified/proc/comfort_owner(datum/source, mob/living/hugger)
+	SIGNAL_HANDLER
+
+	if(is_species(hugger, /datum/species/shadow/nightmare)) //hey wait a minute, that's not a comforting, friendly hug!
+		if(check_surrounding_darkness())
+			addtimer(CALLBACK(src, PROC_REF(freak_out), HUG_TERROR_AMOUNT))
```

Comment:
Do we want this to be a chain-stun? If not, then maybe it will only affect people standing up?

Reply:
That's fair, just wanted to make sure it was thought-out.